File: 03_Lifecycles/01_ViewModel/app/src/main/java/com/codingnight/android/viewmodel/MainActivity.java
package com.codingnight.android.viewmodel;

import androidx.appcompat.app.AppCompatActivity;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;
import android.widget.TextView;

import com.google.android.material.floatingactionbutton.FloatingActionButton;

public class MainActivity extends AppCompatActivity {
    FloatingActionButton fab;
    TextView textView;
    MainViewModel viewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        textView = findViewById(R.id.textView);
        fab = findViewById(R.id.floatingActionButton);

        ViewModelProvider.Factory factory = new MainViewModelFactory();
        viewModel = new ViewModelProvider(this, factory).get(MainViewModel.class);
        textView.setText(String.valueOf(viewModel.getModel().getNumber()));
        fab.setOnClickListener((view) -> {
            viewModel.increment();
            textView.setText(String.valueOf(viewModel.getModel().getNumber()));
        });
    }
}

File: 06_Navigation/06_NavigationDrawer/01_AppBarConfiguration/src/main/java/com/codingnight/example/navigationdrawer/MainActivity.kt
package com.codingnight.example.navigationdrawer

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.navigation.NavController
import androidx.navigation.findNavController
import androidx.navigation.ui.AppBarConfiguration
import androidx.navigation.ui.navigateUp
import androidx.navigation.ui.setupActionBarWithNavController
import androidx.navigation.ui.setupWithNavController
import com.google.android.material.navigation.NavigationView

class MainActivity : AppCompatActivity() {
    private lateinit var appBarConfiguration: AppBarConfiguration
    private lateinit var navController: NavController

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        navController = findNavController(R.id.fragment)

//        appBarConfiguration =
//            01_AppBarConfiguration(navController.graph, findViewById(R.id.drawerLayout))

        appBarConfiguration =
            AppBarConfiguration(
                setOf(
                    R.id.firstFragment,
                    R.id.secondFragment,
                    R.id.thirdFragment
                ), findViewById(R.id.drawerLayout)
            )

        setupActionBarWithNavController(navController, appBarConfiguration)
        (findViewById<NavigationView>(R.id.navigationView)).setupWithNavController(navController)
    }

    override fun onSupportNavigateUp(): Boolean {
        return navController.navigateUp(appBarConfiguration) || super.onSupportNavigateUp()
    }
}